normandy/selfrepair/static/js/normandy_driver.js

```diff
@@ -0,0 +1,85 @@
+/**
+ * Storage class that uses window.localStorage as it's backing store.
+ * @param {string} prefix Prefix to append to all incoming keys.
+ */
+function LocalStorage(prefix) {
+    this.prefix = prefix;
+}
+
+Object.assign(LocalStorage.prototype, {
+    _key(key) {
+        return `${this.prefix}-${key}`;
+    },
+
+    getItem(key) {
+        return new Promise(resolve => {
+            resolve(localStorage.getItem(this._key(key)))
+        });
```

Comment:
You can use [`Promise.resolve`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/resolve) as a shorthand for this:

```javascript
getItem(key) {
    return Promise.resolve(localStorage.getItem(this._key(key));
}
```